runtime/autoload/health/provider.vim

```diff
@@ -487,9 +487,71 @@ function! s:check_ruby() abort
   endif
 endfunction
 
+function! s:check_node() abort
+  call health#report_start('Node provider (optional)')
+
+  let loaded_var = 'g:loaded_node_provider'
+  if exists(loaded_var) && !exists('*provider#node#Call')
+    call health#report_info('Disabled. '.loaded_var.'='.eval(loaded_var))
+    return
+  endif
+
+  if !executable('node') || !executable('npm') || !executable('yarn')
+    call health#report_warn(
+          \ '`node` and `npm` must be in $PATH.',
+          \ ['Install Node.js and verify that `node` and `npm` commands work.'])
+    return
+  endif
+  call health#report_info('Node: '. s:system('node -v'))
+
+  let host = provider#node#Detect()
+  if empty(host)
+    call health#report_warn('Missing "neovim" npm package.',
+          \ ['Run in shell: npm install -g neovim',
+          \  'Is the npm bin directory in $PATH?'])
+    return
+  endif
+  call health#report_info('Host: '. host)
+
+  let latest_npm_cmd = has('win32') ? 'cmd /c npm info neovim --json' : 'npm info neovim --json'
```

Comment:
should not need a separate command for Windows, I believe the ruby check does it that way because of the different quoting. (Though it may possibly be unnecessary there as well, since `system([...])` is called with a list.)

Reply:
Safe to use same command in Windows.

Reply:
Ah, ok.

Reply:
It's the same case for ruby, I think I recall.

Reply:
(This used ruby's provider script as a template)